pkg/manager/manager_upgrade.go

```
@@ -0,0 +1,267 @@
+	rule.ResolvedCommit = latestCommit
+
+	// Re-download the file
+	newURL := strings.Replace(rule.Reference, oldCommit, latestCommit, 1)
+
+	// For each local file, fetch and save the updated content
+	for _, filePath := range rule.LocalFiles {
+		resp, err := http.Get(newURL)
+		if err != nil {
+			return fmt.Errorf("failed to download file: %w", err)
+		}
+		defer resp.Body.Close()
```

golangci-lint (1.64.8)

deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop

(gocritic)